src/cascadia/TerminalApp/App.cpp

```diff
@@ -220,6 +220,25 @@ namespace winrt::TerminalApp::implementation
             const auto& profile = _settings->GetProfiles()[profileIndex];
             auto profileMenuItem = Controls::MenuFlyoutItem{};
 
+            // add the keyboard shortcuts for the first 10 profiles
+            if (profileIndex < 10)
+            {
+                auto profileShortcut = Windows::UI::Xaml::Input::KeyboardAccelerator{};
+                profileShortcut.Modifiers(Windows::System::VirtualKeyModifiers::Control | Windows::System::VirtualKeyModifiers::Shift);
+                
+                if (profileIndex == 9)
+                {
+                    // the last tab is actually 0 in the shortcut, so make sure to handle that
+                    profileShortcut.Key(Windows::System::VirtualKey::Number0);
+                }
+                else

```

Comment:
Is this right? It seems that if you have 9 profiles, only the 9th is added and the others are not.

Reply:
No this works.  The current keybinding has the 10th (profileIndex=9) profile as "0" key shortcut, so I special-cased that one.  It is still iterating through all the profiles (index starting at 0)

Reply:
Woops, my bad. I should have looked at the full code. I didn't realize we are here in a for loop.